### DISCUSSION TITLE
Stick Logger  "Work Mode - Transparency"

### DISCUSSION THREAD

**Discussion by murik85** [davidrapan/ha-solarman#352] [Q&A]
Is it possible to make integration work directly via modbus protocol without solarman api?

![111](https://github.com/user-attachments/assets/fcc946d7-33c0-4c35-96d5-70f3c1849054)

**Comment by davidrapan** [comment]
Proly yes.

**Reply by murik85** [reply]
It's not working.

I understand the integration works over TCP. And Stick Logger's transparent mode requires RTUoverTCP.

Is it possible to add protocol selection to the integration?

**Reply by davidrapan** [reply]
There is Modbus TCP already implemented w/ autodetection. Show the logs.

"It's not working." is never the right answer. 😆

**Reply by murik85** [reply]
[home-assistant_solarman_2025-01-02T20-08-42.806Z.log](https://github.com/user-attachments/files/18294109/home-assistant_solarman_2025-01-02T20-08-42.806Z.log)

**Reply by davidrapan** [reply]
Try to change row 42 in `include/pysolarmanv5/pysolarman.py`:
```
self._passthrough = False
```
to
```
self._passthrough = True
```
and restart HA.

**Reply by murik85** [reply]
[home-assistant_solarman_2025-01-03T05-09-46.278Z.log](https://github.com/user-attachments/files/18296663/home-assistant_solarman_2025-01-03T05-09-46.278Z.log)

**Reply by davidrapan** [reply]
So the stick simply doesn't respond in this transparency mode.

_The integration can comm using modbus tcp w/ passthrough enabled._

**Reply by tccack** [reply]
Hi David... Can you explain what the difference is between the "data collection" and "transparency" work modes on the dongle itself? I am using an LSE-3 but can't write back to the inverter (Deye hybrid). I'm wondering if that mode is responsible?

Tony

**Reply by davidrapan** [reply]
We are not really sure about the difference but it should be set to data collection mode. What do you mean can't write? You need to be more descriptive or provide logs cause it should work.

**Reply by tccack** [reply]
> We are not really sure about the difference but it should be set to data collection mode. What do you mean can't write? You need to be more descriptive or provide logs cause it should work.

I was just interested in the two options in the datalogger interface at the config_hide.html page.
I'm not actually using home assistant but was curious as to why the write back was not (or mostly not working)
To be fair its outside the scope of what you are providing support for but I wondering if I had it set wrong.
(I'm using solar assistant personally)

**Reply by davidrapan** [reply]
I can assure you that using this integration it works and there is nothing you need to setup, switch, enable or whatever 😉

**Reply by tomosvari** [reply]
Hi David, can you a little bit explain me? I use a HYd10 ktl 3ph inverter with an lsw3 datalogger. It functional the last year like a charm, but after the Integrator was updated the firmwares all my data lost and can't even more connect to my logger. I can't see any data. 
HAve build an ESP32-rs485 gateway, it communicat the HA but even no data can see. What am i wrong?
Thanks in advance

**Reply by davidrapan** [reply]
Yeah well, try to check if it's still setup to correct listening port at `http://{logger_ip}/config_hide.html`, but some newer firmware versions have local access removed.

You can read more in https://github.com/davidrapan/ha-solarman/issues/762
